tests/Unit/Domain/Test_BlockNeighbor.cpp

```diff
@@ -0,0 +1,25 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+#include "Domain/BlockNeighbor.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+TEST_CASE("Unit.Domain.BlockNeighbor", "[Domain][Unit]") {
+  // Test default constructor:
+  BlockNeighbor<3> test_blockneighbor;
+  static_cast<void>(test_blockneighbor);
+
+  // Test constructor:
+  Orientation<3> custom_orientation(std::array<Direction<3>, 3>{
+      {Direction<3>::upper_eta(), Direction<3>::upper_zeta(),
+       Direction<3>::upper_xi()}});
+  BlockNeighbor<3> custom_neighbor(0, custom_orientation);
+  CHECK(custom_neighbor.id() == 0);
+  CHECK(custom_neighbor.orientation().mapped(Direction<3>::upper_xi()) ==
+        Direction<3>::upper_eta());
+  CHECK(custom_neighbor.orientation().mapped(Direction<3>::upper_eta()) ==
+        Direction<3>::upper_zeta());
+  CHECK(custom_neighbor.orientation().mapped(Direction<3>::upper_zeta()) ==
+        Direction<3>::upper_xi());
+}
```

Comment:
add checks of equality/inequality operators, test copy/move semantics, test stream operator